py_swf/clients/workflow.py

```diff
@@ -65,3 +66,177 @@ def terminate_workflow(self, workflow_id, reason):
+        :param close_status: string.
+            valid status are ('COMPLETED', 'FAILED', 'CANCELED', 'TERMINATED', 'CONTINUED_AS_NEW', 'TIMED_OUT')
+            This filter has an affect only if executionStatus is specified as CLOSED
+        :return: total number of closed workflows that meet the filter criteria
+        """
+        time_filter_dict = _build_time_filter_dict(
+            oldest_start_date=oldest_start_date,
+            latest_start_date=latest_start_date,
+            oldest_close_date=oldest_close_date,
+            latest_close_date=latest_close_date,
+        )
+
+        count_closed_workflow_within_time_range = partial(
+            self.boto_client.count_closed_workflow_executions,
+            domain=self.workflow_client_config.domain,
+            **time_filter_dict
+        )
+
+        workflow_filter_dict = _build_workflow_filter_dict(
+            workflow_name=workflow_name,
+            version=version,
+            tag=tag,
+            workflow_id=workflow_id,
+            close_status=close_status,
+        )
+
+        return count_closed_workflow_within_time_range(**workflow_filter_dict)
+
+
+def _build_time_filter_dict(oldest_start_date=None, latest_start_date=None, oldest_close_date=None, latest_close_date=None):
+    """
+    Build time_filter_dict for calls to _count_closed_workflow_executions and _count_open_workflow_executions.
+    Return result is a dict: {time_filter_type : filter_dict}
+    filter_dict must contains key 'oldestDate'.
+    return example:
+    {
+        'startTimeFilter': {
+            'oldestDate': datetime(2016, 11, 11)
+        }
+    }
+    If no oldest_date specified, return None
+    """
+    date_tuple = ()
+    if oldest_start_date:
+        filter_type = 'startTimeFilter'
+        date_tuple = (oldest_start_date, latest_start_date)
+    elif oldest_close_date:
+        filter_type = 'closeTimeFilter'
+        date_tuple = (oldest_close_date, latest_close_date)
+
+    if not date_tuple:
+        return None
+
+    oldest_date, latest_date = date_tuple
+    filter_dict = dict(oldestDate=oldest_date)
+    if latest_date:
+        filter_dict['latestDate'] = latest_date
+    return {filter_type: filter_dict}
+
+
+def _build_type_filter_dict(name, version):
+    filter_dict = {'name': name}
+    if version:
+        filter_dict['version'] = version
+    return {'typeFilter': filter_dict}
+
+
+def _build_tag_filter_dict(tag):
+    return {'tagFilter': {'tag': tag}}
+
+
+def _build_execution_filter_dict(workflow_id):
+    return {'executionFilter': {'workflowId': workflow_id}}
+
+
+def _build_close_status_filter_dict(status):
+    return {'closeStatusFilter': {'status': status}}
+
+
+def _build_workflow_filter_dict(
+        workflow_name=None,
+        version=None,
+        tag=None,
+        workflow_id=None,
+        close_status=None,
+):
+    if workflow_name:
```

Comment:
In the docstring of count open/closed workflows there is a description saying that these filters are mutually exclusive.